hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1224,6 +1236,47 @@ def canon(b):
                 full=False,
             )
 
+    def minimize_floats(self):
+        """Some shrinks that we employ that only really make sense for our
+        specific floating point encoding that are hard to discover from any
+        sort of reasonable general principle. This allows us to make
+        transformations like replacing a NaN with an Infinity or replacing
+        a float with its nearest integers that we would otherwise not be
+        able to due to them requiring very specific transformations of
+        the bit sequence.
+
+        We only apply these transformations to blocks that "look like" our
+        standard float encodings because they are only really meaningful
+        there. The logic for detecting this is reasonably precise, but
+        it doesn't matter if it's wrong. These are always valid
+        transformations to make, they
+        """
+        i = 0
+        while i < len(self.shrink_target.examples):
+            ex = self.shrink_target.examples[i]
+            if (
+                ex.label == DRAW_FLOAT_LABEL
+                and len(ex.children) == 2
+                and ex.children[0].length == 8
+            ):
+                u = ex.children[0].start
+                v = ex.children[0].end
+                buf = self.shrink_target.buffer
+                b = buf[u:v]
+                f = lex_to_float(int_from_bytes(b))
+                b2 = int_to_bytes(float_to_lex(f), 8)
+                if b == b2 or self.consider_new_buffer(buf[:u] + b2 + buf[v:]):
```

Comment:
Checking my understanding: If `b != b2` here, does that (usually) imply that `b` is encoded in exponent/mantissa form, and `b2` is the same quantity encoded in integer form?

Reply:
Yes, that's the main way that can happen. I think there are others but I don't remember what they are.